src/Params/PADnoteParameters.h

```diff
@@ -167,12 +176,30 @@ class PADnoteParameters:public Presets
         OscilGen  *oscilgen;
         Resonance *resonance;
 
-        struct Sample {
-            int    size;
+        //! struct containing the sample buffer for one note
+        struct Sample
+        {
+            //! Oscilgen allows params 0..127
+            static constexpr const std::size_t oscilgen_par_levels = 128;
+            //! interpolate between the params with 4 buffers each
+            static constexpr const std::size_t bufs_per_oscilgen_par = 4;
+            //! number of buffers (4 for each oscilgen par)
+            //! will equal smp.size() after allocation
+            static constexpr std::size_t num_buffers() {
+                return oscilgen_par_levels * bufs_per_oscilgen_par;
+            }
+            int    size; //!< size of each sample buffer
             float  basefreq;
-            float *smp;
-        };
 
+            //! pointers to the sample buffers, index is the basewave parameter
+            //! they are all initialized, nullptrs indicate that no wave was calculated for a parameter
+            std::vector<float*> smp;
+            Sample() { smp.resize(num_buffers()); }
+            // copying is deleted at all
+            Sample(const Sample& source) = delete;
+            Sample& operator=(const Sample& source) = delete;
+            Sample& operator=(Sample&& source) = default;
```

Comment:
As long as the copy constructors are correctly handled, then I guess this is alright, though in my experience it's still slightly more error prone than a simple struct with raw pointers (and a bit harder to debug unless you have the gdb scripts to understand std::vectors all working and those tend to break at any non-trivial levels of optimization).